crypto/ec/ec_pmeth.c

```diff
@@ -120,8 +122,19 @@ static int pkey_ec_sign(EVP_PKEY_CTX *ctx, unsigned char *sig, size_t *siglen,
         return 0;
     }
 
-    type = (dctx->md != NULL) ? EVP_MD_type(dctx->md) : NID_sha1;
-
+    type = (dctx->md != NULL) ? EVP_MD_type(dctx->md) : 0;
```

Comment:
Why do you remove the fallback to sha1?

Reply:
In the original source the calling path will be ECDSA_sign -> ECDSA_sign_ex -> kmeth.sign -> ossl_ecdsa_sign  and ossl_ecdsa_sign ignores the parameter `type`
Since I've modified ossl_ecdsa_sign, it will use `type` to setup k_inverse and r. If `type` > 0, the setup method would assume the type as the nid of the hash algorithm.
If it fall backs to sha1 here, when hash algorithm is not set, the deterministic signature may not be correct;
But my method is no correct as well...Maybe I should show an alert or error message here?

Reply:
> If it fall backs to sha1 here, when hash algorithm is not set, the deterministic signature may not be correct;
> But my method is no correct as well...Maybe I should show an alert or error message here?
I'll re-illustrate my words
When (dctx->md==NULL) && (dctx->nonce_type == EVP_PKEY_ECDSA_NONCE_DETERMINISTIC)
I hope it will result to a error message
However, if it fall back to sha1 the setup method would consider it as sha1 determistic signature.

I will change my code to make this not so confusing.